zetta_utils/layer/volumetric/index.py

```diff
@@ -29,13 +29,13 @@ def default_convert(cls, idx_raw: RawVolumetricIndex) -> VolumetricIndex:  # pra
     def to_slices(self):
         return self.bcube.to_slices(self.resolution, self.allow_slice_rounding)
 
-    def pad(self, pad: Vec3D):
+    def pad(self, pad: Union[Vec3D, IntVec3D]):
```

Comment:
How about making a Vec3D generic on the type of tuple element, and then defining a typealias `IntVec3D = Vec3D[int]`? Then `Union[Vec3D, IntVec3D] -> Vec3D`

Reply:
Considered it, but I thought stronger typing was better to prevent mistakes. For instance, there were a lot of places in the code where we were using Vec3D for things like chunk size and stride when we really meant IntVec3D. What you're suggesting is cleaner, though.

I'll see how well mypy does with the type checking when I do that.